### ISSUE TITLE
How are models loaded for evaluation?

### ISSUE THREAD

**Issue by adney11** [Soheil-ab/Orca#24]
Hello @Soheil-ab ,

Thank you for your work!
I'm currently trying to evaluate Orca under different network conditions, but I am unsure where the code loads pre-trained models from.
I see the models being saved to ./train_dir/learner0/model*.ckpt - and I see a load_model and save_model in agent.py - but these functions aren't used anywhere (rephrase: where would be the right place to use load_model)? 
Also - replay_memory has been initiated, but not used - is this on purpose? and How do I continue learning on the previous model using this option?

TL;DR: Could you please explain how to go about loading different models to evaluate Orca's performance?

Thank you!

**Comment by Soheil-ab** [comment]
Hi @adney11,

The code can run in multiple different ways. In general, there are two modes: 
1) You only wanna use a trained model (we call that eval mode).
2) You wanna perform a training session (either from scratch or on top of an existing model, we call that training mode).

That said, if you look at the d5.py script, you'll realize that based on the input arguments, a "learner" can be initiated only when a training session is going to happen. Generally, we don't need a replay buffer to be loaded during the eval scenario. That's why the reply buffer only will be loaded by the learner. 

To see the exact inputs required for each mode/scenario, you can check out the main c code and the lines where we call d5.y script (also, check out the README and the difference b.w. orca.sh and orca-standalone-emulation.sh scripts).

re the models themselves, 
1) The functions you're referring were written as utility functions for other purposes!
2) For saving and loading the models, we simply use some available APIs in TF. In particular, we only pass a so-called checkpoint directory to TF (this is already part of the main json config file). For example, when we are in eval mode, this enforces the agent to load the model and use it. For more details, you can refer to the TF docs e.g. [here](https://www.tensorflow.org/api_docs/python/tf/compat/v1/train/MonitoredTrainingSession).
 
Hope that helps!
